truth_set/java/structure/organisation/dead-code/OrderProcessor.java

```diff
@@ -0,0 +1,95 @@
+import java.util.ArrayList;
+import java.util.List;
+
+
+class OrderProcessor {
+    private List<Order> orders = new ArrayList<>();
+    private double taxRate = 0.07;
+
+    private String errorMessage = "Order was not processed";
+    private Logger logger;
+
+    public OrderProcessor(Logger logger) {
+        this.logger = logger;
+    }
+
+    private double calculateDiscount(double amount) {
+        double discountRate = 0.1;
+        return amount * discountRate;
+    }
+
+    public void processOrder(Order order) {
+        orders.add(order);
+        logger.log("Processing order ID: " + order.getId());
+
+        if (true) {
+            applyTax(order);
+        } else {
+            applyDiscount(order);
+        }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider removing redundant if condition</b></div>
<div id="fix">

Consider removing the redundant `if (true)` condition as it makes the `else` branch unreachable. The `applyTax()` method will always be called while `applyDiscount()` is dead code.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">

```
 @@ -25,9 +25,7 @@
 -        if (true) {
              applyTax(order);
 -        } else {
 -            applyDiscount(order);
 -        }
```




</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/3#issuecomment-2527328141>#55a9ee</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged